### ISSUE TITLE
insert into ... format ORC silently drops rows

### ISSUE THREAD

**Issue by AndLLA** [ClickHouse/ClickHouse#33523]
in the current stable release of clickhouse-client (21.12.3.32),
inserting rows into a table from an ORC file,
fails to insert all rows belonging to the original file, without notice.

previous versions of clickhouse-client used to work flawlessly.

in order to replicate the bug, it is necessary to use an ORC file with more than about 4M rows.

the number of skipped rows is random, about 40% of the total.

**Comment by den-crane** [comment]
@AndLLA  I cannot reproduce with a simple table. Probably the issue is related to datatypes.
Please provide steps to reproduce the issue, create table , orc file,...

```
clickhouse-client -q \
     'select toInt64(number) A, toString(number) S from numbers(100000000) format ORC' > t.orc

drop table if exists trg;
create table trg(A Int64, S String) Engine=MergeTree order by A;

clickhouse-client -q 'insert into trg format ORC' <t.orc

SELECT count() FROM trg
┌───count()─┐
│ 100000000 │
└───────────┘

drop table if exists trg;
create table trg(A Int64, S String) Engine=MergeTree order by A;


clickhouse-client --input_format_parallel_parsing=0 \
        --min_insert_block_size_bytes=0 \
        --min_insert_block_size_rows=1000000000 \
        --max_insert_block_size=1000000000  \
        -q 'insert into trg format ORC' <t.orc   

SELECT count() FROM trg
┌───count()─┐
│ 100000000 │
└───────────┘                       

SELECT
    count(),
    min(rows),
    max(rows),
    sum(rows)
FROM system.parts
WHERE (level = 0) AND (table = 'trg')
┌─count()─┬─min(rows)─┬─max(rows)─┬─sum(rows)─┐
│       1 │ 100000000 │ 100000000 │ 100000000 │
└─────────┴───────────┴───────────┴───────────┘
```

**Comment by Avogar** [comment]
Was fixed in https://github.com/ClickHouse/ClickHouse/pull/32929, will backport this fix to 21.12

**Comment by AndLLA** [comment]
Thanks
